src/components/ConnectionDataTable/index.jsx

```diff
@@ -99,50 +129,83 @@ export default class ConnectionDataTable extends Component {
     };
   }
 
+  handleHeaderClick = sortByHeader => {
+    if (this.props.onHeaderClick) {
+      this.props.onHeaderClick(sortByHeader);
+    }
+  };
+
   render() {
     const {
       classes,
       pageSize,
       columnsSize,
       connection,
       renderRow,
+      headers,
+      sortByHeader,
+      sortDirection,
     } = this.props;
     const { loading } = this.state;
     const { count, page } = this.getPaginationMetadata();
+    const colSpan = columnsSize || (headers && headers.length) || 1;
 
     return (
-      <Table>
-        <TableBody>
-          {connection.edges.length === 0 ? (
-            <TableRow>
-              <TableCell colSpan={columnsSize}>
-                <em>No items for this page.</em>
-              </TableCell>
-            </TableRow>
-          ) : (
-            connection.edges.map(renderRow)
-          )}
-        </TableBody>
-        <TableFooter>
-          <TableRow>
-            {loading ? (
-              <TableCell colSpan={columnsSize} className={classes.loading}>
-                <Spinner size={24} />
-              </TableCell>
-            ) : (
-              <TablePagination
-                colSpan={columnsSize}
-                count={count}
-                labelDisplayedRows={Function.prototype}
-                rowsPerPage={pageSize}
-                rowsPerPageOptions={[pageSize]}
-                page={page}
-                onChangePage={this.handlePageChange}
-              />
+      <div>
```

Comment:
Why multiple divs instead of a Fragment? Why are we rendering 2 tables when loading/why are we changing the existing loading layout?

Reply:
The outermost `div` can definitely be a Fragment but the second `div` is used to keep the table footer from scrolling with the table. To have `TablePagination` not scroll when the table is small as well as the arrows not positioned at the end of the scroll, it needs to be outside the Table component (see [example](https://material-ui-next.com/demos/tables/#sorting-amp-selecting) in material docs). Next, to display the spinner, a `Table` with a `TableFooter` is created in order to have the spinner positioned at the same spot as the arrows in `TablePagination`.

To stop rendering 2 tables, the second Table can be replaced with:

```js
<div className={classes.spinner}>
  <Spinner size={24} />
</div>
```

```js
spinner: {
  height: 56,
  minHeight: 56,
  paddingRight: 2,
  display: 'flex',
  alignItems: 'center',
  flexDirection: 'row-reverse',
},
```
I think ^ makes more sense to use.